Sources/FluentKit/Properties/Field.swift

```diff
@@ -77,15 +77,23 @@ extension FieldProperty: AnyProperty {
     public func output(from output: DatabaseOutput) throws {
         if output.contains([self.key]) {
             self.inputValue = nil
+
+            // decode Value as Optional in case we are nested inside @OptionalGroup
+            // throw FluentError.unexceptedNil if the value is nil
+            let outputValue: Value?
             do {
-                self.outputValue = try output.decode(self.key, as: Value.self)
+                outputValue = try output.decode(self.key, as: Value?.self)
```

Comment:
I'm not sure how decoding an `@OptionalGroup` will work. If the group can have `@OptionalField`s or nested `@OptionalGroup`s inside of it, then how would you tell whether or not the group is optional? 

For example, take this `Fields`:

```
struct Foo: Fields {
    @OptionalField var baz: String?
    @Field var bar: Int
    @OptionalGroup var qux: Qux
}

@OptionalGroup var foo: Foo?
```

When decoding this group, it seems like we should check whether `bar` exists or not to determine whether `foo` is nil. How do we express this in code though. Also, what happens if `Fields` does not contain any non-optional fields?

Reply:
Decoding is done from deepest outwards. And since we don’t re-throw the error this only affects the next OptionalGroup when an nil @Field is found. 
Meaning only OptionalGroups that actually should be nil get set to nil. 

I tried nesting multiple OptionalGroups and seems to work just fine. 

If `Fields` contain no required fields we cannot determine if OptionalGroup should be nil or not resulting it in never being nil. (which is fine i guess. if you don’t have required fields why would one want to use @OptionalGroup regardless?)